File: functions/src/models/onboarding.model.ts
export interface OnboardingQuery {
  search?: string;
  status?: string;
  department?: string;
  branch?: string;
}

export interface OnboardingDocument {
  resume?: string;
  offerLetter?: string;
  aadhaarCard?: string;
  panCard?: string;
  bankProof?: string;
  degreeCertificate?: string;
  passportPhoto?: string;
  experienceCertificate?: string;
  relievingLetter?: string;
}

export interface PersonalDetails {
  fullName: string;
  gender: string;
  dob: string;
  personalEmail: string;
  mobileCountryCode: string;
  mobileNumber: string;
  address: string;
  city: string;
  state: string;
  pincode: string;
  country: string;
  permanentAddress: string;
  emergencyContactName: string;
  emergencyCountryCode: string;
  emergencyContactNumber: string;
}

export interface CompanyDetails {
  companyName: string;
  branchLocation: string;
  dateOfJoining: string;
  department: string;
  designation: string;
  officialEmail?: string;
  employeeId: string;
  shiftTime: string;
  workMode: string;
  employeeType: string;
  experienceLevel: string;
  yearsOfExperience?: number;
  reportingManager: string;
}

export interface BankDetails {
  bankName: string;
  accountHolderName: string;
  accountNumber: string;
  ifscCode: string;
  panNumber: string;
  aadhaarNumber: string;
  pfNumber?: string;
  esiNumber?: string;
  basicSalary: number;
  hra?: number;
  allowances?: number;
  grossSalary: number;
  netSalary: number;
}

export interface EmployeeOnboarding {
  id?: string;
  personalDetails: PersonalDetails;
  companyDetails: CompanyDetails;
  bankDetails: BankDetails;
  documents: OnboardingDocument;
  status: 'pending' | 'approved' | 'rejected' | 'completed';
  createdAt: Date;
  updatedAt?: Date;
}

File: lib/models/onboarding_model.dart
class OnboardingModel {
  final String id;
  final Map<String, dynamic> personalDetails;
  final Map<String, dynamic> companyDetails;
  final Map<String, dynamic> bankDetails;
  final Map<String, dynamic> documents;
  final String status;

  OnboardingModel({
    required this.id,
    required this.personalDetails,
    required this.companyDetails,
    required this.bankDetails,
    required this.documents,
    required this.status,
  });

  factory OnboardingModel.fromJson(Map<String, dynamic> json) {
    return OnboardingModel(
      id: json['id']?.toString() ??
          json['_id']?.toString() ??
          json['documentId']?.toString() ??
          '',
      personalDetails: json['personalDetails'] is Map
          ? Map<String, dynamic>.from(json['personalDetails'] as Map)
          : <String, dynamic>{},
      companyDetails: json['companyDetails'] is Map
          ? Map<String, dynamic>.from(json['companyDetails'] as Map)
          : <String, dynamic>{},
      bankDetails: json['bankDetails'] is Map
          ? Map<String, dynamic>.from(json['bankDetails'] as Map)
          : <String, dynamic>{},
      documents: json['documents'] is Map
          ? Map<String, dynamic>.from(json['documents'] as Map)
          : <String, dynamic>{},
      status: json['status']?.toString() ?? 'pending',
    );
  }
}
